packages/service-provider-core/src/printable-bson.ts

```diff
@@ -65,7 +65,7 @@ export default function(bson): void {
   bson.Int32.prototype.asPrintable = bson.Int32.prototype[toString];
 
   bson.Long.prototype[toString] = function(): string {
-    return `NumberLong(${this.toNumber()})`;
+    return `NumberLong(${this.toString()})`;
```

Comment:
Does it make sense to 

```suggestion
    return `NumberLong("${this.toString()}")`;
```

so that this could then be used as input again without losing preceision?

Reply:
That's brilliant! I think we should do the same for NumberDecimal too then.